Data/HashMap/Base.hs

```diff
@@ -2065,6 +2188,13 @@ updateOrConcatWithKey f ary1 ary2 = A.run $ do
     return mary
 {-# INLINABLE updateOrConcatWithKey #-}
 
+-- | /O(n*m)/ Check if the first array is a subset of the second array.
+subsetArray :: Eq k => (v1 -> v2 -> Bool) -> A.Array (Leaf k v1) -> A.Array (Leaf k v2) -> Bool
+subsetArray cmpV ary1 ary2 = A.length ary1 <= A.length ary2 && A.all inAry2 ary1
+  where
+    inAry2 (L k1 v1) = lookupInArrayCont (\_ -> False) (\v2 _ -> cmpV v1 v2) k1 ary2
+    {-# INLINE inAry2 #-}
```

Comment:
@svenkeidel, @treeowl What did you think about my idea to record the indices of any found elements in `ary2` and to skip these when checking the remaining elements of `ary1`.

I'm mostly wondering whether that's an idea worth recording – I wouldn't insist that it be included in this PR.

Reply:
It is hard to say if this is going to make a big difference performance wise. For each key you still would need to iterate over ary2 anyways. The only thing you would save are a few equality checks of keys. It could be worth working on this optimization, but before we need some benchmarks to verify that we actually improve the performance.

Reply:
@treeowl That's a neat idea! :)

If we don't do this in this PR, I'll record it in an issue.